uefi/flash.go

```diff
@@ -90,6 +101,61 @@ func (f *FlashImage) Validate() []error {
 	return errors
 }
 
+// Extract extracts the flash image to the directory passed in.
+func (f *FlashImage) Extract(dirPath string) error {
+	absDirPath, err := filepath.Abs(dirPath)
+	if err != nil {
+		return err
+	}
+	// Dump the binary
+	f.ExtractPath, err = ExtractBinary(f.buf, absDirPath, "flash.rom")
+	if err != nil {
+		return err
+	}
+
+	// Extract IFD
+	if err = f.IFD.Extract(filepath.Join(absDirPath, "ifd")); err != nil {
```

Comment:
it sure seems like each of these Extract functions has the same type signature. Were you to make Extract an interface, then you could have a nice little struct array with pointers to functions that implement the interface, which these all seem to do, and it would be just a bit tidier (and more friendly to a Fuse implementation later).

Reply:
type Extracter interface {
Extract(dir, base string) error
}

Reply:
Actually Extract is a function in the Firmware Interface shown in uefi.go